api/envoy/extensions/filters/http/mcp/v3/mcp.proto

```diff
@@ -17,3 +18,8 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 message Mcp {
   // TODO: Add configuration fields
 }
+
+// Per-route configuration for MCP filter
+message McpPerRoute {
```

Comment:
If we keep this, consider calling it something like `McpOverride`.  The `PerRoute` suffix is somewhat misleading, since the override config can also be set per-VirtualHost or per-ClusterWeight.